### PR TITLE
[FIX] account: don't skip l10n installation when there's a module l10n_* to be installed

### PR DESCRIPTION
**Pull request by edy1192**
When there's a module to be installed whose name starts with l10n_ but it's not a localization (e.g. `l10n_mx_edi`) it is taken as a localization module so the installation of `l10n_mx` (`country_code = 'MX'`) is skipped.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by edy1192** [comment]
Hi @luisg123v @moylop260 

Could you check this please?

Regards.

**Comment by moylop260** [comment]
@nim-odoo @jco-odoo @smetl 

The domain is getting more cases than expected.

It is using a non-scaped char `_` for sql it means one any char
 - image

It is using odoo `like` operator it means any chars after and before of the value since that this operator adds two `%` e.g. `%value%`:
 - image

So using `l10n\___` and `=like` it will return more precise results
It means, get just modules that starts with `l10n\_` and have any 2 extra chars `__`

Without unexpected corner cases.

**Comment by moylop260** [comment]
For example, `l10n_latam` is not really a `l10n_xx` module.
Or there are `l10n_xx_*` modules that don't depend directly of `l10n_xx`

**Comment by jco-odoo** [comment]
@nhomar @edy1192 l10n_mx_edi would depend on l10n_mx.  The weird thing is that is fix is then only problematic for l10n_mx_edi.  (so, you are hard to follow as I don't see how that is a problem to you)

The latam_base stuff or l10n_be_hr_payroll maybe.  But what do we do with l10n_generic_coa then?

File: addons/account/__init__.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def _auto_install_l10n(cr, registry):
     country_code = env.company.country_id.code
     if country_code:
         #auto install localization module(s) if available
-        to_install_l10n = env['ir.module.module'].search_count([('name', 'like', 'l10n_'), ('state', '=', 'to install')])
+        to_install_l10n = env['ir.module.module'].search_count([('name', '=like', 'l10n\___'), ('state', '=', 'to install')])
```
Comment (luisg123v) [inline addons/account/__init__.py:19]: To consider the module `l10n_generic_coa`, a domain like te following would work IMHO:

```
[
    ('state', '=', 'to install'),
    '|',
    ('name', '=', 'l10n_generic_coa'),
    ('name', '=like', 'l10n\___'),
]
```

Comment (qdp-odoo) [reply addons/account/__init__.py:19]: l10n_syscohada

**Comment by smetl** [comment]
@william-andre Can you investigate the issue there plz? Thanks

**Comment by william-andre** [comment]
The `_auto_install_l10n` serves one main purpose: installing automatically the correct[1] chart of account on the main company.

This is not a correct fix because of these points:
* `l10n_mx_edi` doesn't depend on `l10n_mx`. This would do the trick for your issue.
* modules like `l10n_latam_base` don't depend on any `l10n_` containing a Chart of Accounts. I wouldn't expect to have one installed if we only install a database with this module, especially not `l10n_generic_coa` like it was the case before (and like it would be with your fix)
* we don't actually care about the installation of such modules as we can't (shouldn't) see them from the action `account.open_account_charts_modules` ("Install more packages" from the Accounting settings). (actually, `l10n_latam_base`, `l10n_latam_invoice_document`, `l10n_eu_service` and `l10n_multilang` shoudln't be part of that list).

[1] correct means the one of the country selected in the database manager

**Comment by qdp-odoo** [comment]
on a side note: even though there must be ways to filter out all submodules we don't want, I think it would be cleaner to keep the category "Localization" dedicated to the modules containing the chart of accounts. Other localized stuff should be moved into another category ("accounting" or a new one "localization submodule").

We might do that in master

**Comment by william-andre** [comment]
The issue discussed here will be fixed by https://github.com/odoo/odoo/pull/55384
